scripts/validate_docstrings.py

```diff
@@ -491,12 +492,14 @@ def first_line_ends_in_dot(self):
         if self.doc:
             return self.doc.split('\n')[0][-1] == '.'
 
+    @property
+    def deprecated_with_directive(self):
+        return '.. deprecated:: ' in (self.summary + self.extended_summary)
```

Comment:
if `.. deprecated::` exists in the parameters section, we don't want to consider the docstring as deprecated, that's why it's checking if it exists in the summaries. Is this what you are asking?

Reply:
No my point is that if we expected a very strict placement of the directive after the short but before the extended summary then we wouldn't need to concat the two to get this. Wouldn't it just exist in the summary at that point?

Reply:
That's a good point, and I had to review the code again to remember the reasoning.

For things like ignoring errors in deprecated docstrings, we need to know which functions/methods are deprecated, and this is what `deprecate` returns. But to validate that the `.. deprecated::` directive is placed in the right position, we don't care about all the `Panel` methods, which the script considers deprecated, but only the ones that are deprecated because they have the directive. And this is what `deprecated_with_directive` returns.

@aqurilla it may be worth explaining this briefly in the docstrings of `deprecated` and `deprecated_with_directive`, so it's clear for anyone reading the code.